vendor/github.com/ethereum/go-ethereum/whisper/whisperv5/filter.go

```diff
@@ -115,15 +117,71 @@ func (fs *Filters) NotifyWatchers(env *Envelope, p2pMessage bool) {
 			if match {
 				msg = env.Open(watcher)
 				if msg == nil {
+					if fs.whisper.deliveryServer != nil {
+						err := errors.New("Envelope failed to be opened")
+						switch p2pMessage {
```

Comment:
For `bool` better `if {} else {}`. Also good old way to name booleans is `isP2PMessage`, `doasMatch`, `canDoTricks`. 😄